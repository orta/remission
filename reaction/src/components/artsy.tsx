import * as PropTypes from "prop-types"
import * as React from "react"

/**
 * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextConsumer}
 *
 * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands weâ€™ll be able to not make this
 *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
 */
export interface ContextProps {
  /**
   * The currently signed-in user.
   */
  currentUser?: User
}

interface PrivateContextProps extends ContextProps {
  /**
   * This prop solely exists to verify that a consumer is nested inside a provider.
   */
  _isNestedInProvider: boolean
}

const ContextTypes: React.ValidationMap<PrivateContextProps> = {
  _isNestedInProvider: PropTypes.bool,
  currentUser: PropTypes.object,
}

/**
 * This component provides Artsy specific context props to components down the tree. Components that wish to consume
 * these props should be wrapped with the `ContextConsumer` higher-order component.
 *
 * @see {@link ContextConsumer}
 */
export class ContextProvider extends React.Component<ContextProps, null>
  implements React.ChildContextProvider<PrivateContextProps> {
  static childContextTypes = ContextTypes

  constructor(props) {
    if (React.Children.count(props.children) > 1) {
      throw new Error("A ContextProvider expects a single child.")
    }
    super(props)
  }

  getChildContext() {
    return {
      _isNestedInProvider: true,
      currentUser: this.props.currentUser,
    }
  }

  render() {
    return this.props.children as any
  }
}

/**
 * Wraps a component in a higher-order component that injects the Artsy specific context props into the given component.
 * Wrapped components are expected to be rendered in a tree that contains a `ContextProvider` component.
 *
 * The injected props are described in `ContextProps`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextProps}
 *
 * @param Component The component that needs Artsy specific context props.
 */
export function ContextConsumer<P>(
  Component: React.ComponentClass<P> | React.StatelessComponent<P>
): React.ComponentClass<P> {
  const name = Component.displayName || Component.name
  return class extends React.Component<P, void> {
    static contextTypes = ContextTypes
    static displayName = `Artsy(${name})`

    constructor(props, context: PrivateContextProps) {
      if (!context._isNestedInProvider) {
        const start = name || "A component"
        throw new Error(`${start}, which needs Artsy props, was not wrapped inside a ContextProvider component.`)
      }
      super(props, context)
    }

    render() {
      const currentUser = this.context.currentUser
      const props = Object.assign({ currentUser }, this.props)
      return <Component {...props} />
    }
  }
}
