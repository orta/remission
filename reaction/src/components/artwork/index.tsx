import * as React from "react"
import styled, { css } from "styled-components"

import theme from "../../assets/theme"
import ArtworkMetadata, { ArtworkMetadataProps } from "./metadata"
import createContainer, { RelayProps } from "./relay"

const Container = styled.div`
  width: 100%;
`

const ImageContainer = styled.div`
  width: 100%;
  position: relative;
  cursor: pointer;

  &:before {
    content: '';
    display: block;
    padding-bottom: 100%;
  }

  & .overlay-container {
    transition: opacity 0.25s;

    &.hovered {
      opacity: 0;
      visibility: hidden;
    }

    &.selected {
      opacity: 1;
      visibility: visible;
    }
  }

  &:hover {
    & .overlay-container.hovered {
      ${css`
        @media (min-width: ${theme.flexboxgrid.breakpoints.sm + 1}px) {
          opacity: 1;
          visibility: visible;
        }
      `}
    }
  }
`

const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
  margin: auto;
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
`

export interface OverlayProps {
  selected: boolean
}

export interface ArtworkProps extends RelayProps {
  size?: number
  extended?: boolean
  Overlay?: React.SFC<OverlayProps>
  onSelect?: (selected: boolean) => void
  showOverlayOnHover?: boolean
}

export interface ArtworkState {
  isSelected: boolean
}

export class Artwork extends React.Component<ArtworkProps, ArtworkState> {
  static defaultProps = {
    size: 250,
    extended: true,
    overlay: null,
    showOverlayOnHover: false,
  }

  constructor(props) {
    super(props)
    this.state = {
      isSelected: false,
    }
  }

  onSelected(e) {
    if (!this.props.Overlay) {
      return
    }

    this.setState({
      isSelected: !this.state.isSelected,
    })

    if (this.props.onSelect) {
      this.props.onSelect(!this.state.isSelected)
    }
  }

  render() {
    const { size, artwork, Overlay, showOverlayOnHover } = this.props
    let overlayClasses = "overlay-container"

    overlayClasses += showOverlayOnHover ? " hovered" : ""
    overlayClasses += this.state.isSelected ? " selected" : ""

    return (
      <Container onClick={this.onSelected.bind(this)} size={size}>
        <ImageContainer size={size}>
          <Image src={artwork.image.url} size={size} />
          <div className={overlayClasses}>
            {Overlay && <Overlay selected={this.state.isSelected} />}
          </div>
        </ImageContainer>
        <ArtworkMetadata artwork={artwork} />
      </Container>
    )
  }
}

export default createContainer<ArtworkProps, ArtworkMetadataProps>(Artwork, ArtworkMetadata)
