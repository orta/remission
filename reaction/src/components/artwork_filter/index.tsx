import * as React from "react"
import * as Relay from "react-relay"
import styled from "styled-components"

import Artworks from "../artwork_grid"
import BorderedPulldown from "../bordered_pulldown"
import Spinner from "../spinner"

import Dropdown from "./dropdown"
import ForSaleCheckbox from "./for_sale_checkbox"
import Headline from "./headline"
import TotalCount from "./total_count"

const PageSize = 10

interface Props extends RelayProps, React.HTMLProps<ArtworkFilter> {
  filter_artworks: any
  relay: any
  for_sale?: boolean
  dimension_range?: string
  price_range?: string
  medium?: string
}

interface State {
  for_sale: boolean
  dimension_range: string
  price_range: string
  medium: string
  loading: boolean
}

class ArtworkFilter extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      for_sale: props.for_sale || false,
      dimension_range: props.dimension_range || "*",
      price_range: props.price_range || "*",
      medium: props.medium || "*",
      loading: false,
    }
  }

  handleLoadMore() {
    if (!this.state.loading) {
      this.setState({ loading: true }, () => {
        this.props.relay.setVariables(
          {
            size: this.props.relay.variables.size + PageSize,
          },
          readyState => {
            if (readyState.done) {
              this.setState({ loading: false })
            }
          }
        )
      })
    }
  }

  setForSale() {
    const isForSale = !this.state.for_sale
    const forSaleVar = isForSale ? true : null

    this.setState({
      for_sale: isForSale,
    })
    this.props.relay.setVariables({
      for_sale: forSaleVar,
      size: PageSize,
    })
  }

  onSelect(count, slice) {
    this.setState({
      [slice.toLowerCase()]: count.id,
    })
    this.props.relay.setVariables({
      [slice.toLowerCase()]: count.id,
      size: PageSize,
    })
  }

  onChangeSort(option) {
    this.props.relay.setVariables({
      sort: option.val,
      size: PageSize,
    })
  }

  render() {
    const filterArtworks = this.props.filter_artworks.filter_artworks
    const dropdowns = filterArtworks.aggregations.map(aggregation =>
      <Dropdown
        aggregation={aggregation}
        key={aggregation.slice}
        onSelect={(count, slice) => this.onSelect(count, slice)}
      />
    )
    const pulldownOptions = [
      { val: "-partner_updated_at", name: "Recently Updated" },
      { val: "-year", name: "Artwork Year (desc.)" },
      { val: "year", name: "Artwork Year (asc.)" },
    ]

    return (
      <div>
        <FilterBar>
          <ForSaleCheckbox checked={this.state.for_sale} onClick={() => this.setForSale()} />
          {dropdowns}
        </FilterBar>
        <SubFilterBar>
          <div>
            <Headline
              medium={this.state.medium}
              price_range={this.state.price_range}
              dimension_range={this.state.dimension_range}
              for_sale={this.state.for_sale}
              facet={filterArtworks.facet}
              aggregations={filterArtworks.aggregations}
            />
            <TotalCount filter_artworks={filterArtworks} />
          </div>
          <BorderedPulldown
            defaultValue="Recently Updated"
            options={pulldownOptions}
            onChange={option => this.onChangeSort(option)}
          />
        </SubFilterBar>
        <Artworks artworks={filterArtworks.artworks} onLoadMore={() => this.handleLoadMore()} columnCount={4} />
        <SpinnerContainer>
          {this.state.loading ? <Spinner /> : ""}
        </SpinnerContainer>
      </div>
    )
  }
}

const FilterBar = styled.div`
  text-align: center;
`

const SubFilterBar = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 40px 0 20px;
  align-items: center;
`

const SpinnerContainer = styled.div`
  width: 100%;
  height: 200px;
  position: relative;
`

export default Relay.createContainer(ArtworkFilter, {
  initialVariables: {
    sort: "-partner_updated_at",
    size: PageSize,
    for_sale: null,
    medium: "*",
    aggregations: ["MEDIUM", "TOTAL", "PRICE_RANGE", "DIMENSION_RANGE"],
    price_range: "*",
    dimension_range: "*",
    gene_id: null,
    tag_id: null,
    artist_id: null,
  },
  fragments: {
    filter_artworks: () => Relay.QL`
      fragment on Viewer {
        filter_artworks(
          aggregations: $aggregations, 
          size: $size,
          for_sale: $for_sale,
          medium: $medium,
          price_range: $price_range,
          dimension_range: $dimension_range,
          sort: $sort,
          gene_id: $gene_id,
          tag_id: $tag_id,
          artist_id: $artist_id,
        ) {
          ${TotalCount.getFragment("filter_artworks")}
          aggregations {
            slice 
            counts {
              id
              name
            }
            ${Dropdown.getFragment("aggregation")}
          }
          artworks: artworks_connection(first: $size) {
            ${Artworks.getFragment("artworks")}
          }
          facet {
            ${Headline.getFragment("facet")}
          }
        }
      }
    `,
  },
})

interface RelayProps {
  filter_artworks: {
    filter_artworks: {
      artworks: Array<any | null> | null
      counts: {
        total: number | null
      } | null
      aggregations: Array<any | null> | null
    } | null
  } | null
}
