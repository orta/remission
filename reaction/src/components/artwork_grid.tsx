import * as React from "react"
import * as ReactDOM from "react-dom"
import * as Relay from "react-relay"

import styled from "styled-components"
import Artwork from "./artwork/grid_item"

interface Props extends RelayProps, React.HTMLProps<ArtworkGrid> {
  columnCount?: number
  sectionMargin?: number
  itemMargin?: number
  onLoadMore?: () => any
}

interface State {
  loading: boolean
  interval: any
}

export class ArtworkGrid extends React.Component<Props, State> {
  public static defaultProps: Partial<Props>

  constructor(props) {
    super(props)
    this.state = {
      interval: null,
      loading: false,
    }
  }

  componentDidMount() {
    if (this.props.onLoadMore) {
      const interval = setInterval(() => {
        this.maybeLoadMore()
      }, 150)
      this.setState({ interval })
    }
  }

  componentWillUnmount() {
    if (this.state.interval) {
      clearInterval(this.state.interval)
    }
  }

  maybeLoadMore() {
    const threshold = window.innerHeight + window.scrollY
    const el = ReactDOM.findDOMNode(this)
    if (threshold >= el.clientHeight + el.scrollTop) {
      this.props.onLoadMore()
    }
  }

  sectionedArtworks() {
    const sectionedArtworks: ArtworkRelayProps[][] = []
    const sectionRatioSums = []
    const artworks = this.props.artworks ? this.props.artworks.edges : []

    for (let i = 0; i < this.props.columnCount; i++) {
      sectionedArtworks.push([])
      sectionRatioSums.push(0)
    }

    artworks.forEach(artworkEdge => {
      const artwork = artworkEdge.node

      // There are artworks without images and other ‘issues’. Like Force we’re just going to reject those for now.
      // See: https://github.com/artsy/eigen/issues/1667
      if (artwork.image) {
        // Find section with lowest *inverted* aspect ratio sum, which is the shortest column.
        let lowestRatioSum = Number.MAX_VALUE
        let sectionIndex = null
        for (let j = 0; j < sectionRatioSums.length; j++) {
          const ratioSum = sectionRatioSums[j]
          if (ratioSum < lowestRatioSum) {
            sectionIndex = j
            lowestRatioSum = ratioSum
          }
        }

        if (sectionIndex != null) {
          const section = sectionedArtworks[sectionIndex]
          section.push(artwork)

          // Keep track of total section aspect ratio
          const aspectRatio = artwork.image.aspect_ratio || 1 // Ensure we never divide by null/0
          // Invert the aspect ratio so that a lower value means a shorter section.
          sectionRatioSums[sectionIndex] += 1 / aspectRatio
        }
      }
    })

    return sectionedArtworks
  }

  renderSections() {
    const spacerStyle = {
      height: this.props.itemMargin,
    }
    const sectionedArtworks = this.sectionedArtworks()
    const sections = []

    for (let i = 0; i < this.props.columnCount; i++) {
      const artworkComponents = []
      for (let j = 0; j < sectionedArtworks[i].length; j++) {
        const artwork = sectionedArtworks[i][j]
        artworkComponents.push(<Artwork artwork={artwork as any} key={"artwork-" + j + "-" + artwork.__id} />)
        // Setting a marginBottom on the artwork component didn’t work, so using a spacer view instead.
        if (j < sectionedArtworks[i].length - 1) {
          artworkComponents.push(<div style={spacerStyle} key={"spacer-" + j + "-" + artwork.__id} />)
        }
      }

      const sectionSpecificStyle = {
        flex: 1,
        minWidth: 0,
        marginRight: i === this.props.columnCount - 1 ? 0 : this.props.sectionMargin,
      }

      sections.push(<div style={sectionSpecificStyle} key={i}>{artworkComponents}</div>)
    }
    return sections
  }

  render() {
    const artworks = this.renderSections() || []
    return (
      <div className={this.props.className}>
        {artworks}
      </div>
    )
  }
}

ArtworkGrid.defaultProps = {
  columnCount: 3,
  sectionMargin: 20,
  itemMargin: 20,
}

const StyledGrid = styled(ArtworkGrid)`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`

const ArtworkFragment = Relay.QL`
  fragment on Artwork {
    __id
    image {
      aspect_ratio
    }
    ${Artwork.getFragment("artwork")}
  }
`

export default Relay.createContainer(StyledGrid, {
  fragments: {
    artworks: () => Relay.QL`
      fragment on ArtworkConnection {
        edges {
          node {
            ${ArtworkFragment}
          }
        }
      }
    `,
  },
})

interface ArtworkRelayProps {
  __id: string
  image: {
    aspect_ratio: number | null
  } | null
}

interface RelayProps {
  artworks: {
    edges: Array<{
      node: any
    } | null> | null
  }
}
