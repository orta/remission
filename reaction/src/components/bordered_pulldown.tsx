import * as React from "react"

import Icon from "./icon"

import styled from "styled-components"
import colors from "../assets/colors"
import { secondary } from "../assets/fonts"

interface Props extends React.HTMLProps<BorderedPulldown> {
  options: any
  defaultValue: string
  onChange?: any
}

interface State {
  selected: any
  isHovered: boolean
}

export class BorderedPulldown extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      selected: null,
      isHovered: false,
    }
  }

  toggleHover(value) {
    this.setState({
      isHovered: value,
    })
  }

  onChange(option) {
    this.setState({
      selected: option,
      isHovered: false,
    })
    this.props.onChange(option)
  }

  render() {
    const { options, defaultValue } = this.props

    const optionEls = options.map(option => {
      return (
        <a key={option.val} onClick={() => this.onChange(option)}>
          {option.name}
        </a>
      )
    })

    const displayValue = (this.state.selected && this.state.selected.name) || defaultValue
    let pulldownStyles = {}

    if (this.state.isHovered) {
      pulldownStyles = {
        display: "block",
      }
    }

    return (
      <div
        className={this.props.className}
        onMouseEnter={() => this.toggleHover(true)}
        onMouseLeave={() => this.toggleHover(false)}
      >
        <Toggle>
          <span>
            {displayValue}
          </span>
          <CaretHolder>
            <Icon name="arrow-down" fontSize="9px" color={colors.grayMedium} />
          </CaretHolder>
        </Toggle>
        <PulldownOptions style={pulldownStyles}>
          {optionEls}
        </PulldownOptions>
      </div>
    )
  }
}

const StyledBorderedPulldown = styled(BorderedPulldown)`
  ${secondary.style}
  display: inline-block;
  width: 200px;
  position :relative;
  border: 2px solid ${colors.grayMedium};
  text-align: left;
  font-size: 17px;
  &.is-disabled {
    .bordered-pulldown-toggle {
      cursor: default;
      color: ${colors.grayBold};
    }
  }
`

const Toggle = styled.div`
  display: block;
  padding: 8px 10px 6px;
  text-decoration: none;
`

const CaretHolder = styled.div`
  float: right;
  padding-left: 5px;
  border-left: 1px solid ${colors.grayMedium};
`

const PulldownOptions = styled.div`
  display: none;
  position: absolute;
  background: white;
  border: 2px solid ${colors.grayMedium};
  top: -2px;
  left: -2px;
  right: -2px;
  z-index: 2;
  > a {
    text-decoration: none;
    overflow: ellipsis;
    display: block;
    padding: 8px 10px 6px;
    text-decoration: none;
    cursor: pointer;
    &:hover {
      background-color: ${colors.gray};
    } 
  } 
`

export default StyledBorderedPulldown
