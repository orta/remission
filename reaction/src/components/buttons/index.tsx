import Button from "./default"
import FacebookButton from "./facebook"
import GhostButton from "./ghost"
import InvertedButton from "./inverted"
import TwitterButton from "./twitter"

export default {
  Button,
  FacebookButton,
  GhostButton,
  InvertedButton,
  TwitterButton,
}
