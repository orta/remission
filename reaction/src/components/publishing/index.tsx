import Article from "./article"
import Header from "./header/header"
import RelatedArticlesCanvas from "./related_articles/related_articles_canvas"
import RelatedArticlesPanel from "./related_articles/related_articles_panel"
import Artwork from "./sections/artwork"
import Authors from "./sections/authors"
import Caption from "./sections/caption"
import Embed from "./sections/embed"
import FullscreenViewer from "./sections/fullscreen_viewer/fullscreen_viewer"
import Image from "./sections/image"
import ImageCollection from "./sections/image_collection"
import ImageSetPreview from "./sections/imageset_preview"
import ImageSetPreviewClassic from "./sections/imageset_preview_classic"
import Text from "./sections/text"
import Video from "./sections/video"

// Icon SVGs
import EditEmbed from "./icon/edit_embed"
import EditImages from "./icon/edit_images"
import EditText from "./icon/edit_text"
import EditVideo from "./icon/edit_video"
import HeroImage from "./icon/hero_image"
import HeroVideo from "./icon/hero_video"
import ImageFullscreen from "./icon/image_fullscreen"
import ImageSet from "./icon/image_set"
import LayoutFullscreen from "./icon/layout_fullscreen"
import LayoutSplit from "./icon/layout_split"
import LayoutText from "./icon/layout_text"
import Remove from "./icon/remove"

// Test Fixtures
import { ClassicArticle, FeatureArticle, StandardArticle } from "./fixtures/articles"

export default {
  Article,
  Artwork,
  Authors,
  Embed,
  Caption,
  FullscreenViewer,
  Header,
  Image,
  ImageCollection,
  ImageSetPreview,
  ImageSetPreviewClassic,
  RelatedArticlesCanvas,
  RelatedArticlesPanel,
  Text,
  Video,
  Icon: {
    EditEmbed,
    EditImages,
    EditText,
    EditVideo,
    HeroImage,
    HeroVideo,
    ImageFullscreen,
    ImageSet,
    LayoutFullscreen,
    LayoutSplit,
    LayoutText,
    Remove,
  },
  Fixtures: {
    ClassicArticle,
    FeatureArticle,
    StandardArticle,
  },
}
