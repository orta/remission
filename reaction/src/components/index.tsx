import Artwork from "./artwork"
import ArtworkFilter from "./artwork_filter"
import ArtworkGrid from "./artwork_grid"
import BorderedPulldown from "./bordered_pulldown"
import Buttons from "./buttons/index"
import Checkbox from "./checkbox"
import Gene from "./gene/index"
import Grid from "./grid"
import Icon from "./icon"
import Input from "./input"
import Modal from "./modal/modal"
import Nav from "./nav"
import Text from "./text"
import TextArea from "./text_area"
import TextLink from "./text_link"
import Title from "./title"

import * as React from "react"
import * as ReactDOM from "react-dom"
import * as Relay from "react-relay"

import { artsyNetworkLayer } from "../relay/config"
import * as Artsy from "./artsy"

interface InitOptions {
  user: User
  component: any
  domID: string
  queryConfig: any
}

export function init(options: InitOptions) {
  Relay.injectNetworkLayer(artsyNetworkLayer(options.user))

  const rootRoute = (
    <Artsy.ContextProvider currentUser={options.user}>
      <Relay.RootContainer Component={options.component} route={options.queryConfig} />
    </Artsy.ContextProvider>
  )

  ReactDOM.render(rootRoute, document.getElementById(options.domID))
}

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Gene,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
