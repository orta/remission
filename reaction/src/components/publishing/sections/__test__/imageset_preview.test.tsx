import "jest-styled-components"

import * as React from "react"
import * as renderer from "react-test-renderer"
import { ImageSetFull, ImageSetMini } from "../../fixtures/components"
import ImageSetPreview from "../imageset_preview"

it("renders a full image set properly", () => {
  const imageset = renderer.create(<ImageSetPreview section={ImageSetFull} />).toJSON()
  expect(imageset).toMatchSnapshot()
})
it("renders a mini image set properly", () => {
  const imageset = renderer.create(<ImageSetPreview section={ImageSetMini} />).toJSON()
  expect(imageset).toMatchSnapshot()
})
