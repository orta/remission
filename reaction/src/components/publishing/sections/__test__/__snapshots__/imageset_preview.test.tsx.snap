// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a full image set properly 1`] = `
.c0 {
  position: absolute;
  bottom: 20px;
  left: 20px;
  min-height: 50px;
  width: auto;
  max-width: calc(100% - 80px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid white;
  background: rgba(255, 255, 255, 0.8);
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.3);
  padding: 20px;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  height: 50px;
}

.c2 {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  margin-bottom: 8px;
  line-height: 1.1em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
}

.c5 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin-left: 20px;
}

.c6 {
  height: 45px;
  position: relative;
  margin-left: 40px;
  -webkit-text-align: right;
  text-align: right;
}

.c6 > svg {
  height: 98%;
}

@media (max-width: 600px) {
  .c2 {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

@media (max-width: 600px) {
  .c4 {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
  }
}

@media (max-width: 600px) {
  .c5 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
  }
}

@media (max-width: 600px) {
  .c6 {
    display: none;
  }
}

<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="c0"
    onClick={[Function]}
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        The Work of Bruce M. Sherman
      </div>
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          View Slideshow
        </div>
        <div
          className="c5"
        >
          3
           Images
        </div>
      </div>
    </div>
    <div
      className="c6"
    >
      <svg
        className="image-set"
        viewBox="0 0 44 44"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6760.67,551.72h-32v-32h32v32Zm-30-2h28v-28h-28v28Z"
          transform="translate(-6728.67 -507.72)"
        />
        <polygon
          points="38 37 36 37 36 8 7 8 7 6 38 6 38 37"
        />
        <polygon
          points="44 31 42 31 42 2 13 2 13 0 44 0 44 31"
        />
      </svg>
    </div>
  </div>
  <img
    alt="The Work of Bruce M. Sherman"
    height="auto"
    src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=1200&quality=95"
    width="100%"
  />
</div>
`;

exports[`renders a mini image set properly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  height: 100%;
}

.c0 {
  height: 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 10px 20px 10px 10px;
  border: 1px solid #e5e5e5;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  width: auto;
  margin-left: 20px;
}

.c3 {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  margin-bottom: 8px;
  line-height: 1.1em;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
}

.c6 {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  margin-left: 20px;
}

.c7 {
  height: 45px;
  position: relative;
  margin-left: 40px;
  -webkit-text-align: right;
  text-align: right;
}

.c7 > svg {
  height: 98%;
}

@media (max-width: 600px) {
  .c3 {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

@media (max-width: 600px) {
  .c5 {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
  }
}

@media (max-width: 600px) {
  .c6 {
    font-family: Unica77LLWebRegular ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
  }
}

@media (max-width: 600px) {
  .c7 {
    display: none;
  }
}

<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    className="c0"
    onClick={[Function]}
  >
    <img
      alt="The Work of Bruce M. Sherman"
      height="100%"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=1200&quality=95"
      width="auto"
    />
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          The Work of Bruce M. Sherman
        </div>
        <div
          className="c4"
        >
          <div
            className="c5"
          >
            View Slideshow
          </div>
          <div
            className="c6"
          >
            3
             Images
          </div>
        </div>
      </div>
      <div
        className="c7"
      >
        <svg
          className="image-set"
          viewBox="0 0 44 44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6760.67,551.72h-32v-32h32v32Zm-30-2h28v-28h-28v28Z"
            transform="translate(-6728.67 -507.72)"
          />
          <polygon
            points="38 37 36 37 36 8 7 8 7 6 38 6 38 37"
          />
          <polygon
            points="44 31 42 31 42 2 13 2 13 0 44 0 44 31"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
